docs/configuration.md

```diff
@@ -153,6 +153,12 @@ match:
 # A set of regex-matchers an alert has to fulfill to match the node.
 match_re:
   [ <labelname>: <regex>, ... ]
+  
+# A combined set of equality and regex with support of negative matching 
+# that an alert has to fulfill to match the node.  
+matchers:
+  [ <labelname>: <labelvalue>, <labelname>: <regex>, ... ]
```

Comment:
That's actually not the syntax. From the YAML perspective, `matchers` is just a list of strings. So it should simply be:

```
  [ - <matcher> ... ]
```

(See how it is done below for `routes:`.)

And then we need a separate section below that looks like:

```
## `<matcher>`

A matcher is a string with a syntax inspired by PromQL label matchers.
...
```

Then you have to explain the syntax in detail, which you can compile from https://github.com/prometheus/alertmanager/blob/master/pkg/labels/parse.go#L38-L56 and https://github.com/prometheus/alertmanager/blob/master/pkg/labels/parse.go#L38-L56.

You should probably provide plenty of examples how things look in YAML, including how a single `matcher` can contain many of the matching operations, i.e.
`'{foo = "bar", dings != "bums", }'` but that you can also list them individually as separate list items, e.g.

```YAML
  matchers:
    - '{foo = "bar", dings != "bums", }'
```

is equivalent to

```YAML
  matchers:
    - foo = "bar"
    - dings != "bums"
```

Reply:
Note: I updated the source code links above. (Previously, I gave the same link twice.)

Reply:
Thank you. Will update.

Reply:
> which you can compile from

@beorn7 What does this mean ? Could you elaborate a bit

Reply:
Sorry for using a heavily overloaded word. (o:

That's "compile" in the old sense (before there were compilers): Take the two doc comments I linked above and, based on what's written there, create documentation about the config file (as described above).

(Remember that those that write the config probably aren't aware of the godoc documentation.)

Reply:
Okay got it :) Updated the PR, sorry for the delay. Please feel free to take a look whenever you get some time :)